akka-http-core/src/main/scala/akka/http/javadsl/Http.scala

```diff
@@ -40,6 +42,7 @@ class Http(system: ExtendedActorSystem) extends akka.actor.Extension {
 
   import language.implicitConversions
   private implicit def completionStageCovariant[T, U >: T](in: CompletionStage[T]): CompletionStage[U] = in.asInstanceOf[CompletionStage[U]]
+  @silent("never used")
   private implicit def javaModelIsScalaModel[J <: AnyRef, S <: J](in: Future[J])(implicit ev: JavaMapping.Inherited[J, S]): Future[S] = in.asInstanceOf[Future[S]]
```

Comment:
We could create a method upcast in `Inherited` instead:

```scala
  class Inherited[J <: AnyRef, S <: J](implicit classTag: ClassTag[S]) extends JavaMapping[J, S] {
    def toJava(scalaObject: S): J = scalaObject
    def toScala(javaObject: J): S = cast[S](javaObject)

    import scala.language.higherKinds
    def upcast[F[+_]](f: F[J]): F[S] = f.asInstanceOf[F[S]]
  }
```

Reply:
proposed here: https://github.com/akka/akka-http/pull/2963

Reply:
Nice! That indeed removes this warning entirely, which is of course much nicer than silencing ;)